config/initializers/permissions.rb

```diff
@@ -309,6 +309,34 @@
                      require: :loggedin,
                      dependencies: :view_work_packages
 
+      wpt.permission :view_comments_with_restricted_visibility,
+                     {},
+                     permissible_on: %i[work_package project],
+                     require: :loggedin,
+                     dependencies: :view_work_packages,
+                     visible: -> { OpenProject::FeatureDecisions.comments_with_restricted_visibility_active? }
+
+      wpt.permission :add_comments_with_restricted_visibility,
+                     {},
+                     permissible_on: %i[work_package project],
+                     require: :loggedin,
+                     dependencies: %i[view_work_packages view_comments_with_restricted_visibility],
+                     visible: -> { OpenProject::FeatureDecisions.comments_with_restricted_visibility_active? }
+
+      wpt.permission :edit_own_comments_with_restricted_visibility,
+                     {},
+                     permissible_on: %i[work_package project],
+                     require: :loggedin,
+                     dependencies: %i[view_work_packages view_comments_with_restricted_visibility],
+                     visible: -> { OpenProject::FeatureDecisions.comments_with_restricted_visibility_active? }
+
+      wpt.permission :edit_others_comments_with_restricted_visibility,
+                     {},
+                     permissible_on: %i[work_package project],
+                     require: :loggedin,
+                     dependencies: %i[view_work_packages view_comments_with_restricted_visibility],
```

Comment:
Is the intend to be able to set the permissions on individual work packages? So far, I haven't seen this manifested in the PR, at least in the UI. I see also calls for `User.current.allowed_in_work_package?` which suggests that this is intended. But it seems to me that the permissions are currently only seeded for project permissions. And since there is no UI currently for the work package roles, users will not be able to assign the permission themselves. 

I could see valid use cases for doing this but the reason why I am asking is performance. Querying for work package permissions is way more costly than querying on the project level.

Reply:
Good point. I believe the product requirement is only project permissions, not permission for individual work packages.

We can make the change.

Reply:
This was adjusted and merged into the feature branch. Resolving the comment